File: code_examples/3-Structured_Problem_Solving/18-Nested_Loops/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program uses a nested loop

#include <stdio.h>

int main() {
    // this function uses a nested loop

    // process & output
    for (int counter1 = 0; counter1 < 10; counter1++) {
        for (int counter2 = 0; counter2 < 10; counter2++) {
            printf("Odometer: %d%d\n", counter1, counter2);
        }
    }

    printf("\nDone.\n");
    return 0;
}

File: code_examples/3-Structured_Problem_Solving/18-Nested_Loops/Java/Main.java
/*
 * This program uses a nested loop
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
 */

public class Main {
  public static void main(String[] args) {
    // this function uses a nested loop

    // process & output
    for (int counter1 = 0; counter1 < 10; counter1++) {
      for (int counter2 = 0; counter2 < 10; counter2++) {
        System.out.println("Odometer: %d%d".formatted(counter1, counter2));
      }
    }

    System.out.println("\nDone.");
  }
}
